content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+In the case of a peer to peer network using the IPFS protocol, your computer uses "IPFS-flavored IPLD" to get from raw content to an IPFS address. IPFS has its own preferences and conventions about how data should be broken up into a DAG (more on DAGs below!); IPLD links content on the IPFS network together using those conventions.
+
+**Everything else in the IPFS ecosystem builds on top of this core concept; linked, addressable content is the fundamental connecting element that makes the rest work.**
+
+### 2 \\ DAGs
+
+<img alt="IPFS Stack - Applications" src="../assets/ipfs_stack-apps.png" width="200px" />
+
+IPFS and many other distributed systems take advantage of a concept called [directed acyclic graphs](https://en.wikipedia.org/wiki/Directed_acyclic_graph), or DAGs. DAG structures are what allow distributed nodes to connect to each other in computationally reasonable ways, and are how you can build trees of content-addressed nodes.
+
+As mentioned above, IPFS has its own preferences and conventions about how data should be broken up into DAGs that optimize for its specific use cases. You can do this chunking and structuring in lots of different ways. For example, Git is a DAG that has many versions of your repo inside of it. Its DAG tree doesn't behave in the same way the IPFS DAG does.
+
+IPFS first separates your content into _blocks_. If you had a huge file with a single CID, you'd have to transfer the whole CID every time someone requested it. But Instead, you can break it into blocks, and transfer it block by block instead of one massive chunk at a time. Each block might come from a different source, even! This is a much more efficient way to transmit data. (If you've used BitTorrent, you may have noticed that when you download a file, it fetches it from a bunch of random people; this is the same idea.)
+
+Another great feature of the DAG and breaking content into blocks is that if you have two similar files, they can share the DAG; ie, different trees can reference the same data. For example, if you copyedit a document and change a few commas here and there, IPFS knows to only create new blocks for those small changes. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) much more efficient because you only need to transfer the parts that are new or have changed instead of creating entirely new files each time.
```

Comment:
```suggestion
Another great feature of the DAG and breaking content into blocks is that if you have two similar files, they can share parts of the DAG; ie, parts of different DAGs can reference the same data and thus be the same. For example, if you copyedit a document and change a few commas here and there, IPFS knows to only create new blocks for those small changes. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) much more efficient because you only need to transfer the parts that are new or have changed instead of creating entirely new files each time.
```

Reply:
I combined Hector/Oli suggestions. See what you think.

Reply:
ack